catalog/video/video-admin-plugin/src/test/java/org/codice/alliance/video/ui/TestStreamMonitorHelper.java

```diff
@@ -0,0 +1,141 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+
+public class TestStreamMonitorHelper {
+
+    private StreamMonitorHelper stream;
+
+    private BundleContext bundleContext;
+
+    private class OtherStreamMonitor implements StreamMonitor {
+
+        @Override
+        public Optional<URI> getStreamUri() {
+            return null;
+        }
+
+        @Override
+        public Optional<String> getTitle() {
+            return Optional.of("");
+        }
+
+        @Override
+        public void stopMonitoring() {
+
+        }
+
+        @Override
+        public void startMonitoring() {
+
+        }
+
+        @Override
+        public boolean isMonitoring() {
+            return true;
+        }
+    }
+
+    URI uri;
+
+    @Before
+    public void setUp() throws Exception {
+        bundleContext = mock(BundleContext.class);
+        uri = new URI("udp://localhost:50000");
+        List<ServiceReference<StreamMonitor>> serviceReferences = new ArrayList<>();
+        UdpStreamMonitor udpStreamMonitor = mock(UdpStreamMonitor.class);
+        ServiceReference<StreamMonitor> streamMonitorServiceReference =
+                mock(ServiceReference.class);
+        serviceReferences.add(streamMonitorServiceReference);
+
+        when(bundleContext.getServiceReferences(eq(StreamMonitor.class), anyString())).thenReturn(
+                serviceReferences);
+        when(udpStreamMonitor.getTitle()).thenReturn(Optional.of("test"));
+        when(udpStreamMonitor.getStreamUri()).thenReturn(Optional.of(uri));
+        when(bundleContext.getService(any(ServiceReference.class))).thenReturn(udpStreamMonitor);
+        when(streamMonitorServiceReference.getProperty(anyString())).thenReturn("service.Pid");
+
+        stream = new StreamMonitorHelper(bundleContext);
+    }
+
+    @Test
+    public void testStreamMonitorsNoServiceReferences() throws Exception {
+        List<ServiceReference<StreamMonitor>> serviceReferences = new ArrayList<>();
+        when(bundleContext.getServiceReferences(eq(StreamMonitor.class), anyString())).thenReturn(
+                serviceReferences);
+
+        List<Map<String, Object>> list = stream.udpStreamMonitors();
+        assertThat(list, nullValue());
+    }
+
+    @Test
+    public void testStreamMonistorsWrongService() {
+        when(bundleContext.getService(any(ServiceReference.class))).thenReturn(new OtherStreamMonitor());
+        List<Map<String, Object>> list = stream.udpStreamMonitors();
+        assertThat(list, notNullValue());
+        assertThat(list.size(), is(0));
+    }
+
+    @Test
+    public void testStreamMonitors() {
+        List<Map<String, Object>> list = stream.udpStreamMonitors();
+        assertThat(list, notNullValue());
+        assertThat(list.size(), is(1));
+        Map<String, Object> objectMap = list.get(0);
+        assertThat(objectMap.get("id"), is("service.Pid"));
```

Comment:
`"service.Pid"` could be a constant since you're using it in a few other places.